docs/extending-svd-viewer.md

```diff
@@ -0,0 +1,77 @@
+# Extending SVD Viewer
+
+It is possible to extend the SVD Viewer and register new file extension providers with your VSCode extension. This method will provide reading non-svd formats and load the peripherals information in SVD Viewer. 
+
+## Building your VSCode Extension to extend SVD Viewer
+
+This is a guide about how you can register new peripheral providers to SVD Viewer in your VSCode extension. Please refer to [VSCode Extension API](https://code.visualstudio.com/api) for more information about developing VSCode extensions.
+
+### Adding peripheral-viewer to your VSCode extension
+
+You need to install eclipse-cdt-cloud/vscode-svd-viewer to access the types information. You can use `npm` or `yarn` with the following arguments described below:
+
+Using with npm:
+```bash
+npm install github:eclipse-cdt-cloud/vscode-svd-viewer
+```
+Using with yarn:
+```bash
+yarn add github:eclipse-cdt-cloud/vscode-svd-viewer
+```
+
+### Developing your extension
+
+To provide the peripherals information to SVD Viewer on debug session time, you need register your command which is going construct the peripherals information. The command will receive `DebugSession` object as an input parameter and expects to return array of type `PeripheralOptions[]`.
+
+You can find the example command implementation below:
+
+```js
+import { ExtensionContext } from 'vscode';
+import { 
```

Comment:
Concrete implementations aren't needed, use `type` to ensure none of the extension is included at compile time.

```suggestion
import type { 
```

Reply:
AccessType is an enum so we are unable to define it as `import type` :(